testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy

```diff
@@ -395,6 +404,49 @@ abstract class HttpClientTest extends AgentInstrumentationSpecification {
     method = "HEAD"
   }
 
+  /**
+   * This test fires a large number of concurrent requests.
+   * Each request first hits a HTTP server and then makes another client request.
+   * The goal of this test is to verify that in highly concurrent environment our instrumentations
+   * for http clients (especially inherently concurrent ones, such as Netty or Reactor) correctly
+   * propagate trace context.
+   */
```

Comment:
is the second (downstream) http request needed? is testing concurrency of the initial request enough (under `runUnderTrace` to verify propagation incoming->outgoing)?

also (and maybe this is what I'm missing) why is the test named "causality test" instead of "concurrency test"?

Reply:
"Causality" is due to internal jargon used by @johnbley : we want to check that outgoing request is correctly attributed to the incoming one, hence "causality".

> is the second (downstream) http request needed? is testing concurrency of the initial request enough (under runUnderTrace to verify propagation incoming->outgoing)?

In most cases indeed it would be sufficient to use `runUnderTrace` and to verify that the span, which was active before `doRequest`, is correctly propagated across the wire. But e.g. take an example of async servlets or WebFlux as `/front` request handlers (as separate tests, not part of `HttpClientTest`, but using the same test scenario). In these cases it's obvious where our instrumentation will attribute even the first span. May be it is sufficient to have them as the target of the second request, I don't know...

Reply:
> take an example of async servlets or WebFlux as `/front` request handlers

ya, this makes sense to me, but since these tests don't do this, it's not clear why the extra complexity is needed (though I'm ok with it if you think good to keep it)

Reply:
I have removed the second http call :)